src/main/scala/analysis/SummaryGenerator.scala

```diff
@@ -0,0 +1,186 @@
+) {
+  val lattice: MapLattice[ir.CFGPosition, Set[Taintable], PowersetLattice[Taintable]] = MapLattice(PowersetLattice())
+
+  val domain: Set[CFGPosition] = Set.empty ++ program
+
+  private val stackPointer = Register("R31", 64)
+  private val linkRegister = Register("R30", 64)
+  private val framePointer = Register("R29", 64)
+
+  private val ignoreRegions: Set[Expr] = Set(linkRegister, framePointer, stackPointer)
+
+  def eval(cmd: Command, s: Set[Taintable]): Set[Taintable] = {
+    var m = s
+    val exprs = cmd match {
+      case assume: Assume =>
+        Set(assume.body)
+      case assert: Assert =>
+        Set(assert.body)
+      case memoryAssign: MemoryAssign =>
+        m = m -- getMemoryVariable(cmd, memoryAssign.mem, memoryAssign.index, memoryAssign.size, constProp, globals)
+        Set(memoryAssign.index, memoryAssign.value)
+      case indirectCall: IndirectCall =>
+        if (ignoreRegions.contains(indirectCall.target)) return m
+        Set(indirectCall.target)
+      case assign: Assign =>
+        m = m - assign.lhs
+        Set(assign.rhs)
+      case _ => return m
+    }
+
+    exprs.foldLeft(m) {
+      (m, expr) => {
+        val vars = expr.variables.filter(!ignoreRegions.contains(_)).map { v => v: Taintable }
+        val memvars: Set[Taintable] = expr.loads.flatMap {
+          l => getMemoryVariable(cmd, l.mem, l.index, l.size, constProp, globals)
+        }
+        m.union(vars).union(memvars)
+      }
+    }
+  }
+
+  def transfer(n: CFGPosition, s: Set[Taintable]): Set[Taintable] = n match {
+    case cmd: Command =>
+      eval(cmd, s)
+    case _ => s // ignore other kinds of nodes
+  }
+}
+
+private class RNATaintableSolver(
+  program: Program,
+  globals: Map[BigInt, String],
+  constProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]],
+) extends RNATaintableAnalysis(program, globals, constProp)
+    with IRIntraproceduralBackwardDependencies
+    with Analysis[Map[CFGPosition, Set[Taintable]]]
+    with SimpleWorklistFixpointSolver[CFGPosition, Set[Taintable], PowersetLattice[Taintable]]
+
+/**
+ * Generates summaries (requires and ensures clauses) for procedures that necessarily must hold (assuming a correct implementation).
+ * This helps because the verifier cannot make any assumptions about procedures with no summaries.
+ */
+class SummaryGenerator(
+    program: Program,
+    specGlobals: Set[SpecGlobal],
+    globals: Map[BigInt, String],
+    constProp: Map[CFGPosition, Map[Variable, FlatElement[BitVecLiteral]]],
+    varDepsSummaries: Map[Procedure, Map[Taintable, Set[Taintable]]],
+) {
+  val rnaResults = RNATaintableSolver(program, globals, constProp).analyze()
+
+  // TODO should the stack/link/frame pointers propagate taint?
+  val variables: Set[analysis.Taintable] = 0.to(28).map { n =>
+    Register(s"R$n", 64)
+  }.toSet ++ specGlobals.map { g =>
+    analysis.GlobalVariable(dsl.mem, BitVecLiteral(g.address, 64), g.size, g.name)
+  }
```

Comment:
Only registers R0-R18 and R30 are allowed under the AArch64 standard to change values as the result of a procedure call. This means that there isn't a need to create summaries for the gamma values for R19-R29 and R31 - we already have those.

Reply:
Oh that's helpful, thanks! I was unaware